File: examples/java/kora-java-http-client/src/main/java/ru/tinkoff/kora/example/http/client/RootService.java
package ru.tinkoff.kora.example.http.client;

import ru.tinkoff.kora.common.Component;
import ru.tinkoff.kora.common.annotation.Root;

@Root
@Component
public final class RootService {

    private final InterceptedHttpClient interceptedHttpClient;
    private final JsonHttpClient jsonHttpClient;
    private final MapperRequestHttpClient mapperRequestHttpClient;
    private final MapperResponseHttpClient mapperResponseHttpClient;
    private final ParametersHttpClient parametersHttpClient;
    private final ReactorHttpClient reactorHttpClient;
    private final VoidHttpClient voidHttpClient;
    private final FormHttpClient formHttpClient;

    public RootService(InterceptedHttpClient interceptedHttpClient,
                       JsonHttpClient jsonHttpClient,
                       MapperRequestHttpClient mapperRequestHttpClient,
                       MapperResponseHttpClient mapperResponseHttpClient,
                       ParametersHttpClient parametersHttpClient,
                       ReactorHttpClient reactorHttpClient,
                       VoidHttpClient voidHttpClient,
                       FormHttpClient formHttpClient) {
        this.interceptedHttpClient = interceptedHttpClient;
        this.jsonHttpClient = jsonHttpClient;
        this.mapperRequestHttpClient = mapperRequestHttpClient;
        this.mapperResponseHttpClient = mapperResponseHttpClient;
        this.parametersHttpClient = parametersHttpClient;
        this.reactorHttpClient = reactorHttpClient;
        this.voidHttpClient = voidHttpClient;
        this.formHttpClient = formHttpClient;
    }
}

File: examples/kotlin/kora-kotlin-http-client/src/main/kotlin/ru/tinkoff/kora/kotlin/example/http/client/RootService.kt
package ru.tinkoff.kora.kotlin.example.http.client

import jakarta.annotation.Nullable
import org.slf4j.LoggerFactory
import ru.tinkoff.kora.common.Component
import ru.tinkoff.kora.common.Context
import ru.tinkoff.kora.common.Mapping
import ru.tinkoff.kora.common.annotation.Root
import ru.tinkoff.kora.http.client.common.annotation.HttpClient
import ru.tinkoff.kora.http.client.common.annotation.ResponseCodeMapper
import ru.tinkoff.kora.http.client.common.annotation.ResponseCodeMapper.DEFAULT
import ru.tinkoff.kora.http.client.common.interceptor.HttpClientInterceptor
import ru.tinkoff.kora.http.client.common.request.HttpClientRequest
import ru.tinkoff.kora.http.client.common.request.HttpClientRequestMapper
import ru.tinkoff.kora.http.client.common.response.HttpClientResponse
import ru.tinkoff.kora.http.client.common.response.HttpClientResponseMapper
import ru.tinkoff.kora.http.common.HttpMethod
import ru.tinkoff.kora.http.common.HttpResponseEntity
import ru.tinkoff.kora.http.common.annotation.*
import ru.tinkoff.kora.http.common.body.HttpBody
import ru.tinkoff.kora.http.common.body.HttpBodyOutput
import ru.tinkoff.kora.http.common.form.FormMultipart
import ru.tinkoff.kora.http.common.form.FormUrlEncoded
import ru.tinkoff.kora.json.common.annotation.Json
import java.nio.charset.StandardCharsets
import java.util.concurrent.CompletionStage

@Root
@Component
class RootService(
    private val interceptedHttpClient: InterceptedHttpClient,
    private val jsonHttpClient: JsonHttpClient,
    private val mapperRequestHttpClient: MapperRequestHttpClient,
    private val mapperResponseHttpClient: MapperResponseHttpClient,
    private val parametersHttpClient: ParametersHttpClient,
    private val suspendHttpClient: SuspendHttpClient,
    private val voidHttpClient: VoidHttpClient,
    private val formHttpClient: FormHttpClient
)

